presto-hive/src/main/java/com/facebook/presto/hive/statistics/MetastoreHiveStatisticsProvider.java

```diff
@@ -212,6 +214,34 @@ private Estimate calculateDistinctValuesCount(Map<String, PartitionStatistics> s
                 DoubleStream::max);
     }
 
+    private Estimate calculateDataSize(Map<String, PartitionStatistics> statisticsByPartitionName, String column)
+    {
+        List<Double> knownPartitionDataSizes = statisticsByPartitionName.values().stream()
+                .map(stats -> {
+                    OptionalDouble averageColumnLength = stats.getColumnStatistics().get(column).getAverageColumnLength();
+                    OptionalLong rowCount = stats.getBasicStatistics().getRowCount();
+                    OptionalLong nullsCount = stats.getColumnStatistics().get(column).getNullsCount();
+                    if (!averageColumnLength.isPresent() || !rowCount.isPresent()) {
+                        return OptionalDouble.empty();
+                    }
+
+                    long nonNullsCount = rowCount.getAsLong() - nullsCount.orElse(0);
```

Comment:
`nullsCount.orElse(0)` - This is not correct. If there is no information about number of null - we should return `emtpy`

Reply:
I'm not sure.  I guess it's a question of heuristically is it better to assume a random default datasize, or assume no nulls and just multiply the average column length by the number of nulls.  I also think it's super unlikely that there will be a situation where you have the average column length, but not the number of nulls, but that's beside the point.

Reply:
From my perspective if you don't know the number of nulls it is very similar to the situation when you don't know the number of rows. Null fraction can be pretty high, so the size estimate may be way off. I don't have a strong opinion though.